database-plugins/src/main/java/co/cask/hydrator/plugin/teradata/batch/source/TeradataSource.java

```diff
@@ -0,0 +1,145 @@
+    "will be replaced by 'splitBy' field limits specified by the bounding query.";
+  private static final String BOUNDING_QUERY_DESCRIPTION = "Bounding Query should return the min and max of the " +
+    "values of the 'splitBy' field. For example, 'SELECT MIN(id),MAX(id) FROM table'";
+  private static final String SPLIT_FIELD_DESCRIPTION = "Field Name which will be used to generate splits.";
+
+  private final TeradataSourceConfig sourceConfig;
+  private final DBManager dbManager;
+  private Class<? extends Driver> driverClass;
+
+  public TeradataSource(TeradataSourceConfig sourceConfig) {
+    this.sourceConfig = sourceConfig;
+    this.dbManager = new DBManager(sourceConfig);
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    dbManager.validateJDBCPluginPipeline(pipelineConfigurer, getJDBCPluginId());
+    Preconditions.checkArgument(sourceConfig.importQuery.contains("$CONDITIONS"), "Import Query %s must contain the " +
+      "string '$CONDITIONS'.", sourceConfig.importQuery);
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    LOG.debug("pluginType = {}; pluginName = {}; connectionString = {}; importQuery = {}; " +
+                "boundingQuery = {}",
+              sourceConfig.jdbcPluginType, sourceConfig.jdbcPluginName,
+              sourceConfig.connectionString, sourceConfig.importQuery, sourceConfig.boundingQuery);
+
+    Job job = Job.getInstance();
+    Configuration hConf = job.getConfiguration();
+    hConf.clear();
+
+    // Load the plugin class to make sure it is available.
+    Class<? extends Driver> driverClass = context.loadPluginClass(getJDBCPluginId());
+    if (sourceConfig.user == null && sourceConfig.password == null) {
+      DBConfiguration.configureDB(hConf, driverClass.getName(), sourceConfig.connectionString);
+    } else {
+      DBConfiguration.configureDB(hConf, driverClass.getName(), sourceConfig.connectionString,
+                                  sourceConfig.user, sourceConfig.password);
+    }
+    DataDrivenETLDBInputFormat.setInput(job, DBRecord.class, sourceConfig.importQuery, sourceConfig.boundingQuery);
+    job.getConfiguration().set(DBConfiguration.INPUT_ORDER_BY_PROPERTY, sourceConfig.splitBy);
+    context.setInput(new SourceInputFormatProvider(DataDrivenETLDBInputFormat.class, hConf));
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    super.initialize(context);
+    driverClass = context.loadPluginClass(getJDBCPluginId());
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, DBRecord> input, Emitter<StructuredRecord> emitter) throws Exception {
+    emitter.emit(StructuredRecordUtils.convertCase(
+      input.getValue().getRecord(), FieldCase.toFieldCase(sourceConfig.columnNameCase)));
+  }
+
+  @Override
+  public void destroy() {
+    DBUtils.cleanup(driverClass);
```

Comment:
can this throw any exceptions? If so should catch and make sure destroy is called on dbManager.